geoips/commandline/geoips_command.py

```diff
@@ -303,59 +331,192 @@ def _get_registry_by_interface_and_package(self, interface, package_name):
-                        product_dict[subplg_name]["relpath"],  # Relpath
-                    ]
-                    table_data.append(plugin_entry)
-            else:
-                plugin_entry = [
-                    interface_registry[plugin_key]["package"],
-                    interface.name,
-                    interface.interface_type,
-                    interface_registry[plugin_key]["family"],
-                    plugin_key,
-                    "N/A",
-                    interface_registry[plugin_key]["relpath"],
-                ]
-                table_data.append(plugin_entry)
-        headers = [
-            "GeoIPS Package",
-            "Interface",
-            "Interface Type",
-            "Family",
-            "Plugin Name",
-            "Source Names",
-            "Relative Path",
-        ]
+        default_headers = {
+            "package": "GeoIPS Package",
+            "interface": "Interface Name",
+            "plugin_type": "Interface Type",
+            "family": "Family",
+            "plugin_name": "Plugin Name",
+            "source_names": "Source Names",
+            "relpath": "Relative Path",
+        }
+        headers = self._get_headers_by_command(args, default_headers)
+        table_data = self._generate_table_data_by_interface(
+            interface,
+            interface_registry,
+            headers,
+        )
         print(
             tabulate(
                 table_data,
-                headers=headers,
+                headers=headers.values(),
                 tablefmt="rounded_grid",
                 maxcolwidths=self.terminal_width // len(headers),
             )
         )
+
+    def _generate_table_data_by_interface(self, interface, interface_registry, headers):
+        """Generate the table data needed for output in '_print_plugins_short_format.
+
+        Given a certain interface and a list of headers, generate a list of table data
+        used for outputting information used by 'geoips list interface' and
+        'geoips list plugins'.
+
+        Parameters
+        ----------
+        interface: geoips.interfaces.<interface_type>
+            - The interface object we will be gathering data from.
+        interface_registry: dict
+            - The plugin registry associated with the given interface.
+        headers: dict
+            - Dictionary of strings representing the key-value mapping of the
+              headers we'd like to output.
+
+        Returns
+        -------
+        table_data: 2D List of Strings
+            - The corresponding data we will output, where each sub-list is ordered
+              by headers.
+        """
+        table_data = []
+        if interface.name == "products":
+            for plugin_key in sorted(interface_registry.keys()):
+                product_dict = interface_registry[plugin_key]
+                table_data += self._get_entry(product_dict, headers)
```

Comment:
This is making me realize that we're not handling yaml plugins whose families are "list" correctly. Currently the only interface for which we implement lists of plugins is the Products interface, however, I always intended for the "list" family to be usable by any yaml-based plugin interface. So we could have lists of sectors, for instance.

This means that we should be handling plugins whose family is "list" more generally and they should be handled in the plugin registry rather than needing to be handled by the CLI. I'm not sure what the correct way to handle this is, but it's something that we need to work on. The difficulty with the Product plugins is that the same product name can exist for multiple source_names.

This is something that we should discuss and come up with a solution for.

Reply:
This will be fixed in #478 which includes update to how we access product plugins. Internally we'll search for them as a plugin name of ``source_name.product_name``, and access them via ``get_plugin`` by ``products.get_plugin(source_name, product_name)``. If we do this for every yaml that is of family ``list``, there shouldn't be a problem in the future.

Reply:
Got it, thanks!